cluster-provision/gocli/cmd/provision.go

```diff
@@ -274,6 +275,11 @@ func provisionCluster(cmd *cobra.Command, args []string) (retErr error) {
 	if err != nil {
 		return err
 	}
+	timeout := time.Second * 10
```

Comment:
Thanks, but i am not sure that we don't workaround some other problem with it,
we have on line R269 `ssh.sh sudo shutdown now -h` which stops the VM, once the VM is stopped, as it was the entrypoint of the container, the container will stop by itself as far as i remember,
adding a stop means we intervene with it, if we can debug it on CI once it happens, to understand why the VM hangs might be better?

even if at the end we take it, maybe worth to check what is a legitimate timeout, maybe 10 seconds is not enough for the VM to gracefully shutdown,
we also need to make sure that it doesn't have consequences (well, i believe that there is a problem with some rpm/kernel that is hidden, and even if we dont add the stop might appear sometimes, for example if node get restarted, which we don't usually do, so in this aspect i guess we are fine)

Reply:
> Thanks, but i am not sure that we don't workaround some other problem with it, we have on line R269 `ssh.sh sudo shutdown now -h` which stops the VM, once the VM is stopped, as it was the entrypoint of the container, the container will stop by itself as far as i remember, adding a stop means we intervene with it, if we can debug it on CI once it happens, to understand why the VM hangs might be better?

I haven't been unable to reproduce this issue locally but when I did catch it occurring on the cluster - I did not see anything in the vm console that indicated that the shutdown failed to complete. However the qemu process does seem to hang.
> 
> i wonder if the stop can hang as well by itself when the problem occurs, unless it is `stop --force`
> 
> even if at the end we take it, maybe worth to check what is a legitimate timeout, maybe 10 seconds is not enough for the VM to gracefully shutdown, we also need to make sure that it doesn't have consequences (well, i believe that there is a problem with some rpm/kernel that is hidden, and even if we dont add the stop might appear sometimes, for example if node get restarted, which we don't usually do, so in this aspect i guess we are fine)
> 
> maybe we can enable some logging, that if it hangs, then using the stop, we would able to see the log when it starts again during `cluster-up` ? also a console during the shutdown can show why it is hanged maybe
> 
> EDIT - having said all this, did you see it improves the failures ? if so, just worth to takes it and at least give more timeout, and maybe enable some logging so we would able to see on the next boot if some problem occurred maybe? (assuming the stop doesn't hang)

I ran this a number of times against the lanes and they all completed successfully. I think it is worth a try - it shouldn't cause any harm anyway.